server/src/main/java/org/opentosca/toscana/plugins/cloudformation/CloudFormationFileCreator.java

```diff
@@ -0,0 +1,174 @@
+        }
+    }
+
+    /**
+     * Creates all Scripts necessary for AWS CloudFormation deployment.
+     */
+    public void createScripts() throws IOException {
+        createFileUploadScript();
+        createDeployScript();
+    }
+
+    /**
+     * Creates a deploy script for deploying the cloudformation template.
+     */
+    private void createDeployScript() throws IOException {
+        // TODO maybe add the execution of the fileUploadScript
+        logger.debug("Creating deploy script.");
+        BashScript deployScript = new BashScript(cfnModule.getFileAccess(), FILENAME_DEPLOY);
+        deployScript.append(EnvironmentCheck.checkEnvironment("aws"));
+        deployScript.append(CHANGE_TO_PARENT_DIRECTORY);
+        deployScript.append(CLI_COMMAND_CREATESTACK
+            + CLI_PARAM_STACKNAME + cfnModule.getStackName() + " "
+            + CLI_PARAM_TEMPLATEFILE + TEMPLATE_PATH);
+    }
+
+    /**
+     * Creates the script for File Uploads if files need to be uploaded.
+     */
+    private void createFileUploadScript() throws IOException {
+        ArrayList<String> filesToBeUploaded = cfnModule.getFilesToBeUploaded();
+
+        logger.debug("Checking if files need to be uploaded.");
+        if (!filesToBeUploaded.isEmpty()) {
+            logger.debug("Files to be uploaded found.");
+            logger.debug("Creating file upload script.");
+            BashScript fileUploadScript = new BashScript(cfnModule.getFileAccess(), FILENAME_UPLOAD);
+            fileUploadScript.append(createBucket());
+            
+            logger.debug("Adding file upload commands.");
+            filesToBeUploaded.forEach((filePath) -> {
+                String localFilePath = RELATIVE_DIRECTORY_PREFIX + filePath;
+                try {
+                    fileUploadScript.append(uploadFile(filePath, localFilePath));
+                } catch (IOException e) {
+                    logger.error("Adding file uploads failed.");
+                    logger.error("See the stack trace for more info.");
+                    e.printStackTrace();
+                }
+            });
+        } else {
+            logger.debug("No files to be uploaded found.");
+            logger.debug("Skipping creation of file upload script.");
+        }
+    }
+
+    /**
+     * Creates an S3Bucket with the given name.
+     * Wraps resources/cloudformation.scripts/create-bucket.sh
+     */
+    private String createBucket() {
+        // TODO get region, currently defaults to us-west-2
+        return "createBucket \"" + cfnModule.getBucketName() + "\"" + " \"us-west-2\"";
+    }
+
+    /**
+     * Puts the given file with the given key on the S3Bucket with  the given name.
+     * Wraps resources/cloudformation.scripts/upload-file.sh
+     */
+    private String uploadFile(String objectKey, String filePath) {
+        return "uploadFile \"" + cfnModule.getBucketName() + "\" \"" + objectKey + "\" \"" + filePath + "\"";
```

Comment:
Don't know whether you actually need the `"` here

Reply:
Going to check whether this is needed

Reply:
After thinking a bit about this, this is just the normal passing of String parameters in bash.
The `""` are used for Strings that have spaces in them. Therefore you don't need them for the bucketName or any other parameters that do not have spaces in them. 
For objectKey and filePath: I believe that these could possibly contain spaces in their paths, so I'll be keeping the `""` around them.

Reply:
After thinking a bit about this, I think this is just the normal passing of String parameters.
The "" are used for Strings that have spaces in them. Therefore you don't need them for the bucketName or any other parameters that do not have spaces in them.
For objectKey and filePath: I believe that these could possibly contain spaces in their paths, so I'll be keeping the "" around them.